code.js

```diff
@@ -1,3 +1,62 @@
-function dijkstra(graph, sourceNode) {
-    return [];
+function initializeDAlgorithm(graph){
+	visitedNodes={}
+	keyValues=Object.keys(graph);
+	//console.log(keyValues.length)
+	if (keyValues.length == 0)
+		return {}
+	if (keyValues.length == 1){
+		visitedNodes[keyValues[0]] = 0
+		//console.log(visitedNodes)
+		return visitedNodes
+	}
+	source=keyValues[0]
+	for (var i=0; i<Object.keys(graph).length; i++){
+		visitedNodes[keyValues[i]]= Infinity
+	}
+	visitedNodes[source]=0
+	dAlgorithm(graph, 0, source);
+	return visitedNodes
 }
+
+function dAlgorithm(graph, currentDistance, currentNode){
+	allVisited=true;
+	visitedKeys=Object.keys(visitedNodes)
+	for (var i=0; i<visitedKeys.length; i++){
+		if (visitedNodes[visitedKeys[i]]===Infinity)
+			allVisited=false;
+	}
+	if (allVisited==true){
+		return visitedNodes
+	}
+	smallestWeight=Infinity;
+	smallestEdge=null
+	keyValues=Object.keys(graph);
+        for (var j=0; j < graph[currentNode].length; j++){
+		if (visitedNodes[graph[currentNode][j].node]===Infinity){
```

Comment:
Not sure why you need to check this condition here -- Dijkstra's algorithm doesn't do it.